### PR TITLE
*: Enable tiflash_fastscan variable and remove fast mode alter statment

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve?

Issue Number: ref #35850

Problem Summary:

1. remove the code related to the ddl 'alter table xx set tiflash mode fast/normal'.
2. add session/global variable tiflash_fastscan to control whether use fast scan in tiflash.
3. change the telemetry statistic for fast scan.

[Should merge after the PR(https://github.com/pingcap/tipb/pull/272/files) merged]

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/find_best_task.go [hunk 1]
```diff
@@ -1888,7 +1888,7 @@ func (ds *DataSource) convertToTableScan(prop *property.PhysicalProperty, candid
 		return invalidTask, nil
 	}
 	ts, cost, _ := ds.getOriginalPhysicalTableScan(prop, candidate.path, candidate.isMatchProp)
-	if ts.KeepOrder && ts.StoreType == kv.TiFlash && (ts.Desc || ts.Table.TiFlashMode == model.TiFlashModeFast) {
+	if ts.KeepOrder && ts.StoreType == kv.TiFlash && (ts.Desc || ds.SCtx().GetSessionVars().TiFlashFastScan) {
```
Comment (breezewish) [inline planner/core/find_best_task.go:1891]: What will happen in this case?

File: sessionctx/variable/tidb_vars.go [hunk 2]
```diff
@@ -231,6 +231,9 @@ const (
 
 	// RequireSecureTransport indicates the secure mode for data transport
 	RequireSecureTransport = "require_secure_transport"
+
+	// TiFlashFastScan indicates whether use fast scan in tiflash .
+	TiFlashFastScan = "tiflash_fastscan"
```
Comment (breezewish) [inline sessionctx/variable/tidb_vars.go:236]: TiFlashEnableFastScan?

**Review by breezewish — COMMENTED** [review]
the rest lgtm

File: planner/core/find_best_task.go (hunk 1 above)
Comment (hongyunyan) [reply planner/core/find_best_task.go:1891]: If the plan set keepOrder = true and TiflashFastScan = true, we will regard the plan is invalid, and return directly.

File: sessionctx/variable/tidb_vars.go (hunk 2 above)
Comment (hongyunyan) [reply sessionctx/variable/tidb_vars.go:236]: I think it should keep the same with the session/global variable.

**Review by AstroProfundis — APPROVED** [review]
The telemetry part LGTM

**Review by breezewish — APPROVED** [review]

File: telemetry/data_window_test.go [hunk 3]
```diff
@@ -40,3 +47,57 @@ func TestBuiltinFunctionsUsage(t *testing.T) {
...
+				peer2 := c.AllocID()
+				addr2 := fmt.Sprintf("tiflash%d", tiflashIdx)
+				mockCluster.AddStore(store2, addr2, &metapb.StoreLabel{Key: "engine", Value: "tiflash"})
+				mockCluster.AddPeer(region1, store2, peer2)
+				tiflashIdx++
+			}
+		}),
+		mockstore.WithStoreType(mockstore.EmbedUnistore),
+	)
+}
```
Comment (xhebox) [inline telemetry/data_window_test.go:69]: Add an empty line here.

**Review by xhebox — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Comment by solotzg** [comment]
@hongyunyan PTAL https://ci.pingcap.net/blue/organizations/jenkins/tiflash-ghpr-integration-tests/detail/tiflash-ghpr-integration-tests/8414/pipeline

```
[2022-08-29T09:53:59.517Z]   File: fullstack-test2/ddl/alter_table_tiflash_replica_and_mode.test

[2022-08-29T09:53:59.517Z]   Error line: 35

[2022-08-29T09:53:59.517Z]   Error: alter table test.t set tiflash mode fast

[2022-08-29T09:53:59.517Z]   Result:

[2022-08-29T09:53:59.517Z]     ERROR 1064 (42000) at line 1: You have an error in your SQL syntax; check the manual that corresponds to your TiDB version for the right syntax to use line 1 column 35 near "mode fast"

[2022-08-29T09:53:59.517Z]   Expected:

[2022-08-29T09:53:59.517Z]     <nothing>

```

**Pull request merged**